### PR TITLE
feat(elements|ino-chip): add slot for trailing icon

### LINKED ISSUE #204: ino-chip: allow custom trailing icons
Currently, it is only possible to position a custom icon at the leading position of an `ino-chip`. We should provide more flexibility so the consumer is able to position the icon at the trailing position as well.

DoD's:
- Custom icons can be placed at the trailing position (default should be leading)
- The trailing icon does not collide with the `inoRemovable` option (maybe the removable option will be ignored if a trailing icon is provided?)

### PR DESCRIPTION
**Pull request by P1NHE4D**
✨ **Features:**
* Added `ino-icon-leading` slot to the `ino-chip` component

🎟️ **Closes #204**

### CODE REVIEW

File: packages/elements/src/components/ino-chip/ino-chip.scss [hunk 1]
```diff
@@ -15,6 +15,18 @@ ino-chip {
   .mdc-chip {
     @include chips.fill-color-accessible($default-chip-color);
 
+    .mdc-chip__icon--trailing {
+      width: 20px;
+      height: 20px;
+      font-size: 20px;
+    }
+
+    ino-icon {
```
Comment (janivo) [inline packages/elements/src/components/ino-chip/ino-chip.scss:24]: This seems to be a duplication of lines 24 - 28, no?

File: packages/elements/src/components/ino-chip/ino-chip.scss [hunk 2]
```diff
@@ -15,6 +15,18 @@ ino-chip {
   .mdc-chip {
     @include chips.fill-color-accessible($default-chip-color);
 
+    .mdc-chip__icon--trailing {
```
Comment (janivo) [inline packages/elements/src/components/ino-chip/ino-chip.scss:18]: Why is this necessary? Do the leading and trailing icons not have the same size?

File: packages/elements/src/components/ino-chip/ino-chip.tsx [hunk 3]
```diff
@@ -36,6 +36,11 @@ export class Chip implements ComponentInterface {
    */
   @Prop() inoIconLeading = false;
 
+  /**
+   * If enabled, appends the slotted icon to the chip label
```
Comment (janivo) [inline packages/elements/src/components/ino-chip/ino-chip.tsx:40]: We also should mention that this flag will disable the `ino-removable` functionality

**Review by janivo — CHANGES_REQUESTED** [review]

**Comment by janivo** [comment]
Hey @P1NHE4D, could you have a look at this again?

**Comment by P1NHE4D** [comment]
> Hey @P1NHE4D, could you have a look at this again?

I'm on it right now. 🙂

File: packages/elements/src/components/ino-chip/ino-chip.scss (hunk 1 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-chip/ino-chip.scss:24]: apparently 😄

File: packages/elements/src/components/ino-chip/ino-chip.tsx (hunk 3 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-chip/ino-chip.tsx:49]: ✅

File: packages/elements/src/components/ino-chip/ino-chip.scss (hunk 2 above)
Comment (P1NHE4D) [reply packages/elements/src/components/ino-chip/ino-chip.scss:18]: Removing the class results in trailing icons being smaller than leading icons. We're also using the 'mdc-chip__icon--leading' class for the leading icon. However, due to the absence of a corresponding class for trailing icons in the mdc package, I added the 'mdc-chip__icon--trailing' class manually. I also attached a screenshot to illustrate the effect of removing the aforementioned.
Skjermbilde 2021-02-03 kl  10 50 49

Comment (janivo) [reply packages/elements/src/components/ino-chip/ino-chip.scss:18]: Alright, thanks for the explanation ✔️

**Review by janivo — CHANGES_REQUESTED** [review]
Overall this seems to collide with the changes made in https://github.com/inovex/elements/pull/277 as we removed all `inoIconLeading/Trailing` slots and check for them at runtime. 
Could you rebase on the #277 and update the render function accordingly?

**Comment by janivo** [comment]
Ah, my bad here. I thought something went wrong while rebasing on master, so I tried it myself and force-pushed. I realized that I told you to rebase on the #277 and that it's not merged yet 🤦🏼‍♂️

**Comment by janivo** [comment]
I'll review as soon as #277 is merged

**Comment by janivo** [comment]
@P1NHE4D could you rebase on last time? 😅

**Review by janivo — APPROVED** [review]
looking good!
